finality-aleph/src/party/manager/task.rs

```diff
@@ -47,3 +48,34 @@ impl Task {
         result
     }
 }
+
+/// Common args for tasks.
+#[derive(Clone)]
+pub struct TaskCommon {
+    pub spawn_handle: SpawnHandle,
+    pub session_id: u32,
+}
+
+#[async_trait::async_trait]
+pub trait Runnable: Send + 'static {
+    async fn run(self, exit: oneshot::Receiver<()>);
+}
+
+/// Runs the given task within a single session.
+pub fn task<R: Runnable>(subtask_common: TaskCommon, runnable: R, name: &'static str) -> Task {
```

Comment:
nit: I would put the name parameter inside the `Runnable` trait but it is purely cosmetic, feel free to ifnore

Reply:
Hmm, I thought about this a bit, but I think it's better to leave the naming decision to the invoking party. It did make me notice one silly thing in nearby code though. <_<"